common/src/androidTest/java/com/microsoft/identity/common/crypto/AndroidWrappedKeyLoaderTest.java

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.crypto;
+
+import android.content.Context;
+import android.security.KeyPairGeneratorSpec;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationSettings;
+import com.microsoft.identity.common.internal.util.FileUtil;
+import com.microsoft.identity.common.internal.util.AndroidKeyStoreUtil;
+import com.microsoft.identity.common.java.crypto.key.AES256KeyLoader;
+import com.microsoft.identity.common.java.exception.ClientException;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.math.BigInteger;
+import java.security.KeyPair;
+import java.security.spec.AlgorithmParameterSpec;
+import java.util.Date;
+
+import javax.crypto.SecretKey;
+import javax.security.auth.x500.X500Principal;
+
+public class AndroidWrappedKeyLoaderTest {
+
+    final Context context = InstrumentationRegistry.getInstrumentation().getTargetContext();
+
+    @Before
+    public void setUp() throws Exception {
+        // Everything is on clean slate.
+        AuthenticationSettings.INSTANCE.clearSecretKeysForTestCases();
+        AndroidKeyStoreUtil.deleteKey(AndroidWrappedKeyLoader.KEY_STORE_CERT_ALIAS);
+        FileUtil.deleteFile(context, AndroidWrappedKeyLoader.KEY_FILE_PATH);
+    }
+
+    @Test
+    public void testKeyStoreOperations() throws ClientException {
+        final String alias = "SOME_ALIAS";
+        final String keyAlgo = "RSA";
+        final String cipherAlgo = "RSA/ECB/PKCS1Padding";
```

Comment:
probably can define as constants at top of class in case these need to be used in additional tests

Reply:
for constants... it sounds like we might want to exercise some different options here.  Should we parameterize this test?